src/sbi_cove.adoc

```diff
@@ -761,30 +769,66 @@ enum tsm_state {
     TSM_READY = 2
 };
 
+#define COVE_TSM_CAP_PROMOTE_TO_TVM       0x1
+#define COVE_TSM_CAP_LOCAL_ATTESTATION    0x2
+#define COVE_TSM_CAP_AIA                  0x3
+#define COVE_TSM_CAP_DYN_PAGE_CONVERSION  0x4
+
 struct tsm_info {
     /*
-     * The current state of the TSM (see tsm_state enum above).
-     * If the state is not TSM_READY, the remaining fields are invalid and will
-     * be initialized to 0.
+     * The current state of the TSM (see `tsm_state` enum above).
+     * If the state is not `TSM_READY`, the remaining fields are invalid and
+     * will be initialized to `0`.
      */
     uint32_t tsm_state;
-    /* Version number of the running TSM. */
+    /* 
+     * Identifier of the TSM implementation. 
+     * It is intended to distinguish among different TSM implementations,
+     * potentially managed by different organizations, that might target
+     * different deployment models and, thus, implement subset of CoVE spec.
+     */
+    uint32_t tsm_impl_id;
+    /* 
+     * Version number of the running TSM. 
+     */
     uint32_t tsm_version;
+    /* 
+     * CoVE features supported by the running TSM.
+     * Enabled bit `i` indicates that the TSM and underly supports capability `i` and there
+     * is hardware support.
+     */
+    uint64_t tsm_capabilities;
     /*
      * The number of 4KiB pages which must be donated to the TSM for storing TVM
-     * state in sbi_covh_create_tvm_vcpu().
+     * state in sbi_covh_create_tvm_vcpu(). `0` if the TSM does not support the
+     * dynamic page conversion capability.
      */
-    unsigned long tvm_state_pages;
-    /* The maximum number of vCPUs a TVM can support. */
-    unsigned long tvm_max_vcpus;
+    uint64_t tvm_state_pages;
+    /* 
+     * The maximum number of vCPUs a TVM can support. 
+     */
+    uint64_t tvm_max_vcpus;
     /*
-     * The number of 4KB pages which must be donated to the TSM when
-     * creating a new vCPU.
+     * The number of 4KiB pages which must be donated to the TSM when creating
+     * a new vCPU. `0` if the TSM does not support the dynamic page conversion
+     * capability.
      */
-    unsigned long tvm_vcpu_state_pages;
+    uint64_t tvm_vcpu_state_pages;
```

Comment:
Any specific reasons to change these to uint64_t ? In the SBI spec, we try to use unsigned long if other data type is not warranted. This allows the spec to easier to implement on RV32 (probably not applicable in this case) or future RV128. 

There are instances where uint64_t or uint32_t usage in the spec. But those had clear requirements.

Reply:
Thanks, I changed these types back to `unsigned long`.